ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/keys/KeyMonotonicityValidationSpec.scala

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.kvutils.committer.transaction.keys
+
+import java.time.{Instant, ZoneOffset, ZonedDateTime}
+
+import com.daml.ledger.participant.state.kvutils.Conversions
+import com.daml.ledger.participant.state.kvutils.DamlKvutils._
+import com.daml.ledger.participant.state.kvutils.committer.StepContinue
+import com.daml.ledger.participant.state.kvutils.committer.transaction.TransactionCommitter
+import com.daml.ledger.participant.state.kvutils.committer.transaction.TransactionCommitter.DamlTransactionEntrySummary
+import com.daml.ledger.participant.state.v1.RejectionReason
+import com.google.protobuf.ByteString
+import org.mockito.{ArgumentMatchersSugar, MockitoSugar}
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AsyncWordSpec
+
+class KeyMonotonicityValidationSpec
+    extends AsyncWordSpec
+    with Matchers
+    with MockitoSugar
+    with ArgumentMatchersSugar {
+
+  private val testKey = DamlStateKey.newBuilder().build()
+  private val testSubmissionSeed = ByteString.copyFromUtf8("a" * 32)
+  private val baseTime = ZonedDateTime.of(2021, 1, 1, 12, 0, 0, 0, ZoneOffset.UTC).toInstant
+  private val testTransactionEntry = {
+    DamlTransactionEntrySummary(
+      DamlTransactionEntry
+        .newBuilder()
+        .setSubmissionSeed(testSubmissionSeed)
+        .setLedgerEffectiveTime(Conversions.buildTimestamp(baseTime))
+        .build()
+    )
+  }
+
+  "checkContractKeysCausalMonotonicity" should {
+    "create StepContinue in case of correct keys" in {
+      KeyMonotonicityValidation.checkContractKeysCausalMonotonicity(
+        mock[TransactionCommitter],
+        None,
+        Set(testKey),
+        Map(testKey -> buildStateValue(baseTime.minusSeconds(1))),
```

Comment:
You may improve test readability by renaming the builder method to something like `aStateValueActiveAt` or similar, plus renaming `baseTime` to `ledgerEffectiveTime` or `ledgerTime`. This way, it would be obvious that activeness is being tweaked:
```scala
  ...
  Map(testKey -> aStateValueActiveAt(ledgerEffectiveTime.minusSeconds(1))),
  ...
```